### PR TITLE
pipelinePlanner

### PR DESCRIPTION
**Pull request by berg0401**
@captain-yoshi 

Creating a MTC pipelineplanner with a BT leaf node

### CODE REVIEW

File: src/create_mtc_pipeline_planner.cpp [hunk 1]
```diff
@@ -31,38 +35,61 @@ BT::NodeStatus CreateMTCPipelinePlanner::tick()
...
+    BT::InputPort<double>(kPortGoalJointTolerance, "1e-4", "tolerance for reaching joint goals"),
+    BT::InputPort<double>(kPortGoalPositionTolerance, "1e-4", "tolerance for reaching position goals"),
+    BT::InputPort<double>(kPortGoalOrientationTolerance, "1e-4", "tolerance for reaching orientation goals"),
+    BT::InputPort<bool>(kPortDisplayMotionPlans, false, "..."),
+    BT::InputPort<bool>(kPortPublishPlanningRequests, false, "..."),
+    BT::InputPort<std::string>(kPortPlannerID, "planner's id"),
+    BT::InputPort<std::string>(kPortPipelineID, "pipeline's id"),
+    BT::InputPort<uint>(kPortNumPlanningAttempts, "1u", "number of planning attempts"),
+    BT::InputPort<double>(kPortMaxVelocityScalingFactor, "maximum velocity allowed"),
+    BT::InputPort<double>(kPortMaxAccelerationScalingFactor, "maximum velocity allowed"),
```
Comment (captain-yoshi) [inline src/create_mtc_pipeline_planner.cpp]: Let's set the default as 0.1 and reuse the same description as MTC.
```suggestion
    BT::InputPort<double>(kPortMaxVelocityScalingFactor, 0.1, "scale down max velocity by this factor"),
    BT::InputPort<double>(kPortMaxAccelerationScalingFactor, 0.1, "scale down max acceleration by this factor"),
```

File: src/create_mtc_pipeline_planner.cpp [hunk 2]
```diff
@@ -31,38 +35,61 @@ BT::NodeStatus CreateMTCPipelinePlanner::tick()
...
+    //Output
+    BT::OutputPort<MTC::solvers::PlannerInterfacePtr>(kPortSolver, "{solver}", "Planner interface using pipeline motion solver"),
+    //Inputs
+    BT::InputPort<double>(kPortGoalJointTolerance, "1e-4", "tolerance for reaching joint goals"),
+    BT::InputPort<double>(kPortGoalPositionTolerance, "1e-4", "tolerance for reaching position goals"),
+    BT::InputPort<double>(kPortGoalOrientationTolerance, "1e-4", "tolerance for reaching orientation goals"),
+    BT::InputPort<bool>(kPortDisplayMotionPlans, false, "..."),
+    BT::InputPort<bool>(kPortPublishPlanningRequests, false, "..."),
+    BT::InputPort<std::string>(kPortPlannerID, "planner's id"),
+    BT::InputPort<std::string>(kPortPipelineID, "pipeline's id"),
```
Comment (captain-yoshi) [inline src/create_mtc_pipeline_planner.cpp]: No need to add a description as it is pretty self explanatory. At least for seasoned MoveIt users ;)
```suggestion
    BT::InputPort<std::string>(kPortPlannerID),
    BT::InputPort<std::string>(kPortPipelineID),
```

File: src/create_mtc_pipeline_planner.cpp [hunk 3]
```diff
@@ -31,38 +35,61 @@ BT::NodeStatus CreateMTCPipelinePlanner::tick()
...
+    return NodeStatus::FAILURE;
+  if(!getInput(kPortMaxAccelerationScalingFactor, max_acceleration_scaling_factor))
+    return NodeStatus::FAILURE;
+  //Optionnal inputs
+  getInput(kPortGoalJointTolerance, goal_joint_tolerance);
+  getInput(kPortGoalPositionTolerance, goal_position_tolerance);
+  getInput(kPortGoalOrientationTolerance, goal_orientation_tolerance);
+  getInput(kPortDisplayMotionPlans, display_motion_plans);
+  getInput(kPortPublishPlanningRequests, publish_planning_requests);
+  getInput(kPortNumPlanningAttempts, num_planning_attemps);
```
Comment (captain-yoshi) [inline src/create_mtc_pipeline_planner.cpp]: Remove comments `//ex : ...`, the second one is not valid.

For the time being, let's make all inputs mandatory. Rework all the getInputs like the following:
```cpp
if(!getInput(kPortPipelineID, pipeline_id) ||
   !getInput(kPortPlannerID, planner_id) ||
   ...
   !getInput(kPortNumPlanningAttempts, num_planning_attemps))
      return NodeStatus::FAILURE;
```

**Review by captain-yoshi — COMMENTED** [review]
Almost there! Please update to reflect the changes in the review.

File: src/create_mtc_pipeline_planner.cpp [hunk 4]
```diff
@@ -31,38 +35,61 @@ BT::NodeStatus CreateMTCPipelinePlanner::tick()
...
-    BT::InputPort<double>(kPortMaxAccelerationScalingFactor, 0.1, "cale down max acceleration by this factor"),
-    BT::OutputPort<MTC::solvers::PlannerInterfacePtr>(kPortSolver),
+    //Output
+    BT::OutputPort<MTC::solvers::PlannerInterfacePtr>(kPortSolver, "{solver}", "Planner interface using pipeline motion solver"),
+    //Inputs
+    BT::InputPort<double>(kPortGoalJointTolerance, "1e-4", "tolerance for reaching joint goals"),
+    BT::InputPort<double>(kPortGoalPositionTolerance, "1e-4", "tolerance for reaching position goals"),
+    BT::InputPort<double>(kPortGoalOrientationTolerance, "1e-4", "tolerance for reaching orientation goals"),
+    BT::InputPort<bool>(kPortDisplayMotionPlans, false, "..."),
+    BT::InputPort<bool>(kPortPublishPlanningRequests, false, "..."),
```
Comment (captain-yoshi) [inline src/create_mtc_pipeline_planner.cpp]:
```suggestion
    BT::InputPort<bool>(kPortDisplayMotionPlans, false, 
                                     "publish generated solutions on topic " + planning_pipeline::PlanningPipeline::DISPLAY_PATH_TOPIC),
    BT::InputPort<bool>(kPortPublishPlanningRequests, false, 
    "publish motion planning requests on topic " + planning_pipeline::PlanningPipeline::MOTION_PLAN_REQUEST_TOPIC),
```

**Comment by berg0401** [comment]
@captain-yoshi clang-format added

**Pull request merged**